packages/core-app-api/src/app/AppManager.test.tsx

```diff
@@ -569,4 +570,150 @@ describe('Integration Test', () => {
       ),
     ]);
   });
+
+  describe('relative url resolvers', () => {
+    const checkConfigValue = async (
+      data: object,
+      configString: string,
+      expectedValue: string,
+    ) => {
+      const app = new AppManager({
+        apis: [],
+        defaultApis: [],
+        themes: [
+          {
+            id: 'light',
+            title: 'Light Theme',
+            variant: 'light',
+            Provider: ({ children }) => <>{children}</>,
+          },
+        ],
+        icons,
+        plugins: [],
+        components,
+        configLoader: async () => [
+          {
+            context: 'test',
+            data,
+          },
+        ],
+      });
+
+      const Provider = app.getProvider();
+      const ConfigDisplay = () => {
+        const apiHolder = useApiHolder();
+        const config = apiHolder.get(configApiRef);
+        return <span>{config?.getString(configString)}</span>;
+      };
+
+      const dom = await renderWithEffects(
+        <Provider>
+          <ConfigDisplay />
+        </Provider>,
+      );
+
+      // Verify that the backend.baseUrl is set correctly by the AppManager.
+      expect(dom.getByText(expectedValue)).toBeTruthy();
+    };
+    [
+      {
+        data: {
+          backend: {
+            baseUrl: 'http://localhost:8008/',
+          },
+          app: {
+            baseUrl: 'http://localhost:8008/',
+          },
+        },
+        paths: ['app.baseUrl', 'backend.baseUrl'],
+      },
+      {
+        data: {
+          backend: {
+            baseUrl: 'http://test.com/',
+          },
+          app: {
+            baseUrl: 'http://test.com/',
+          },
+        },
+        paths: ['app.baseUrl', 'backend.baseUrl'],
+      },
+    ].forEach(item => {
+      item.paths.forEach(path => {
+        it('should force the urls to be relative when they are the same', async () => {
+          await checkConfigValue(item.data, path, 'http://localhost/');
+        });
```

Comment:
Let's avoid wrapping `it` inside logic. I think in this case you can probably refactor this to just be a single `it.each` that covers all the cases.